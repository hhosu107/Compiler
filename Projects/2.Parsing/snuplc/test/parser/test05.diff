20,21c20
<           neg <int>
<             2147483648 <int>
---
>           -2147483648 <int>
