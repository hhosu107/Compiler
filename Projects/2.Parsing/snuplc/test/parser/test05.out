parsing 'test05.mod'...
successfully parsed.
  AST:
    CAstScope: 'test05'
      symbol table:
        [[
          [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
          [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
          [ *ReadInt() --> <int>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<int>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
          [ @max      <int>           ]
          [ @min      <int>           ]
        ]]
      statement list:
        := <int>
          [ @min      <int> ] <int>
          neg <int>
            2147483648 <int>
        := <int>
          [ @max      <int> ] <int>
          2147483647 <int>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest05.mod.ast.pdf test05.mod.ast.dot


Done.
